File: modulo_11_billetera_multimodal/internal/domain/transaction.go
package domain

import (
	"time"

	"github.com/google/uuid"
)

type Transaction struct {
	ID            uuid.UUID  `json:"id"`
	WalletID      uuid.UUID  `json:"wallet_id"`
	Type          string     `json:"type"`
	Amount        float64    `json:"amount"`
	BalanceBefore float64    `json:"balance_before"`
	BalanceAfter  float64    `json:"balance_after"`
	Description   string     `json:"description"`
	ReferenceID   *uuid.UUID `json:"reference_id,omitempty"`
	Status        string     `json:"status"`
	CreatedAt     time.Time  `json:"created_at"`
}

type TransactionType string

const (
	TransactionTypeRecharge  TransactionType = "RECHARGE"
	TransactionTypePayment   TransactionType = "PAYMENT"
	TransactionTypeTransfer  TransactionType = "TRANSFER"
	TransactionTypeRefund    TransactionType = "REFUND"
)

type TransactionStatus string

const (
	TransactionStatusPending   TransactionStatus = "PENDING"
	TransactionStatusCompleted TransactionStatus = "COMPLETED"
	TransactionStatusFailed    TransactionStatus = "FAILED"
)

File: modulo_12_api_publica/backend/src/domain/transaction.ts
export interface Transaction {
  id: string;
  wallet_id: string;
  type: 'RECHARGE' | 'PAYMENT' | 'TRANSFER' | 'REFUND';
  amount: number;
  balance_before: number;
  balance_after: number;
  description: string | null;
  reference_id: string | null;
  status: 'PENDING' | 'COMPLETED' | 'FAILED';
  created_at: Date;
}
